# __ANALISI DEL TASSO DI DISOCCUPAZIONE GIOVANILE IN ITALIA NEGLI ANNI 2010 - 2019__

**Esercizio laboratorio per i punti aggiuntivi all'esame di Alice Ilacqua, Laura Sorrentino e Colombo Rebecca**

Il dataset fa riferimento al tasso di disoccupazione giovanile italiana negli anni 2010 - 2019 raggruppata per età 15-24 e 25-34 anni; tale dataset verrà confrontato con quello del tasso di occupazione giovanile.


Il dataset utilizzato si può trovare sul sito ISTAT cliccando il seguente link: http://dati-giovani.istat.it/ PERCORSO DA SEGUIRE PER VISUALIZZARE TABELLA: Lavoro--Disoccupazione--Tasso di disoccupazione--Popolazione di 15-34 anni - reg.-- selezionare periodo 2010 - 2019 togliendo i trimestri --selezionare fasce età 15-24 e 25-34)

In [17]:
import numpy as np
import pandas as pd

In [18]:
tabella_iniziale_disoc= pd.read_csv("C:/Users/lalla/Desktop/UNIMI/Tasso disoccupazione.csv")
tabella_iniziale_disoc

ITTER107 Territorio TIPO_DATO_FOL                Tipo dato  SEXISTAT1  \
0        ITC1   Piemonte        UNEM_R  tasso di disoccupazione          1   
1        ITC1   Piemonte        UNEM_R  tasso di disoccupazione          1   
2        ITC1   Piemonte        UNEM_R  tasso di disoccupazione          1   
3        ITC1   Piemonte        UNEM_R  tasso di disoccupazione          1   
4        ITC1   Piemonte        UNEM_R  tasso di disoccupazione          1   
...       ...        ...           ...                      ...        ...   
1315     ITE3     Marche        UNEM_R  tasso di disoccupazione          2   
1316     ITE3     Marche        UNEM_R  tasso di disoccupazione          2   
1317     ITE3     Marche        UNEM_R  tasso di disoccupazione          2   
1318     ITE3     Marche        UNEM_R  tasso di disoccupazione          2   
1319     ITE3     Marche        UNEM_R  tasso di disoccupazione          2   

        Sesso    ETA1 Classe di età  TIME  Seleziona periodo      Value  \
0      maschi  Y15-24    15-24 anni  2010               2010  26.669293   
1      maschi  Y15-24    15-24 anni  2011               2011  23.777270   
2      maschi  Y15-24    15-24 anni  2012               2012  29.870147   
3      maschi  Y15-24    15-24 anni  2013               2013  40.233077   
4      maschi  Y15-24    15-24 anni  2014               2014  41.636268   
...       ...     ...           ...   ...                ...        ...   
1315  femmine  Y15-24    15-24 anni  2015               2015  36.901490   
1316  femmine  Y15-24    15-24 anni  2016               2016  37.043544   
1317  femmine  Y15-24    15-24 anni  2017               2017  27.681294   
1318  femmine  Y15-24    15-24 anni  2018               2018  28.532478   
1319  femmine  Y15-24    15-24 anni  2019               2019  32.129859   

      Flag Codes  Flags  
0            NaN    NaN  
1            NaN    NaN  
2            NaN    NaN  
3            NaN    NaN  
4            NaN    NaN  
...          ...    ...  
1315         NaN    NaN  
1316         NaN    NaN  
1317         NaN    NaN  
1318         NaN    NaN  
1319         NaN    NaN  

[1320 rows x 13 columns]

In [9]:
tabella_iniziale_disoc.isnull().sum()

ITTER107                0
Territorio              0
TIPO_DATO_FOL           0
Tipo dato               0
SEXISTAT1               0
Sesso                   0
ETA1                    0
Classe di età           0
TIME                    0
Seleziona periodo       0
Value                   0
Flag Codes           1320
Flags                1320
dtype: int64

**Tutti i valori di "Flag Codes" e "Flag" sono nulli (1320) pertanto eliminiamo le due colonne perchè non ci interessano per la nostra analisi e creiamo un nuovo dataset con le colonne che ci interessano**

In [10]:
tasso_disoccupazione_giovanile= pd.DataFrame(data=tabella_iniziale_disoc, columns =['Territorio', 'SEXISTAT1', 'Sesso', 'Classe di età', 'Seleziona periodo', 'Value' ])
tasso_disoccupazione_giovanile

Territorio  SEXISTAT1    Sesso Classe di età  Seleziona periodo  \
0      Piemonte          1   maschi    15-24 anni               2010   
1      Piemonte          1   maschi    15-24 anni               2011   
2      Piemonte          1   maschi    15-24 anni               2012   
3      Piemonte          1   maschi    15-24 anni               2013   
4      Piemonte          1   maschi    15-24 anni               2014   
...         ...        ...      ...           ...                ...   
1315     Marche          2  femmine    15-24 anni               2015   
1316     Marche          2  femmine    15-24 anni               2016   
1317     Marche          2  femmine    15-24 anni               2017   
1318     Marche          2  femmine    15-24 anni               2018   
1319     Marche          2  femmine    15-24 anni               2019   

          Value  
0     26.669293  
1     23.777270  
2     29.870147  
3     40.233077  
4     41.636268  
...         ...  
1315  36.901490  
1316  37.043544  
1317  27.681294  
1318  28.532478  
1319  32.129859  

[1320 rows x 6 columns]

**La tabella non ci evidenzia la fascia di età 25-34 ma in realtà i valori ci sono (vedi sotto):**

In [19]:
Territori=tasso_disoccupazione_giovanile.groupby(["Classe di età"]).count().reset_index()
Territori["Classe di età"]

0    15-24 anni
1    25-34 anni
Name: Classe di età, dtype: object

In [12]:
Territori=tasso_disoccupazione_giovanile.groupby(["Territorio"]).count().reset_index()
Territori["Territorio"]

0                                Abruzzo
1                             Basilicata
2                               Calabria
3                               Campania
4                         Emilia-Romagna
5                  Friuli-Venezia Giulia
6                                  Lazio
7                                Liguria
8                              Lombardia
9                                 Marche
10                                Molise
11                              Piemonte
12    Provincia Autonoma Bolzano / Bozen
13             Provincia Autonoma Trento
14                                Puglia
15                              Sardegna
16                               Sicilia
17                               Toscana
18        Trentino Alto Adige / Südtirol
19                                Umbria
20        Valle d'Aosta / Vallée d'Aoste
21                                Veneto
Name: Territorio, dtype: object

Come si può notare, le due Province di Bolzano e Trento rientrano nella regione del Trentino Alto Adige quindi le cancelliamo dal dataset:

In [14]:
#TDG="tasso disoccupazione giovanile"
TDG_regioni=tasso_disoccupazione_giovanile[~tasso_disoccupazione_giovanile.Territorio.str.contains("Provincia")]
TDG_regioni

Territorio  SEXISTAT1    Sesso Classe di età  Seleziona periodo  \
0      Piemonte          1   maschi    15-24 anni               2010   
1      Piemonte          1   maschi    15-24 anni               2011   
2      Piemonte          1   maschi    15-24 anni               2012   
3      Piemonte          1   maschi    15-24 anni               2013   
4      Piemonte          1   maschi    15-24 anni               2014   
...         ...        ...      ...           ...                ...   
1315     Marche          2  femmine    15-24 anni               2015   
1316     Marche          2  femmine    15-24 anni               2016   
1317     Marche          2  femmine    15-24 anni               2017   
1318     Marche          2  femmine    15-24 anni               2018   
1319     Marche          2  femmine    15-24 anni               2019   

          Value  
0     26.669293  
1     23.777270  
2     29.870147  
3     40.233077  
4     41.636268  
...         ...  
1315  36.901490  
1316  37.043544  
1317  27.681294  
1318  28.532478  
1319  32.129859  

[1200 rows x 6 columns]

**Il mio dataset adesso è TDG_regioni quindi poi posso creare altri dataset partendo da questo**

In [15]:
primo_data=TDG_regioni[['Territorio', 'Sesso', 'Value']]
primo_data

Territorio    Sesso      Value
0      Piemonte   maschi  26.669293
1      Piemonte   maschi  23.777270
2      Piemonte   maschi  29.870147
3      Piemonte   maschi  40.233077
4      Piemonte   maschi  41.636268
...         ...      ...        ...
1315     Marche  femmine  36.901490
1316     Marche  femmine  37.043544
1317     Marche  femmine  27.681294
1318     Marche  femmine  28.532478
1319     Marche  femmine  32.129859

[1200 rows x 3 columns]

In [16]:

#trovo i totali sui 9 anni
primo_data=primo_data.groupby(["Territorio", "Sesso"]).sum()
primo_data

Value
Territorio                     Sesso              
Abruzzo                        femmine  619.909845
                               maschi   463.910165
                               totale   524.645933
Basilicata                     femmine  720.903003
                               maschi   600.648645
                               totale   645.900615
Calabria                       femmine  925.583371
                               maschi   777.642595
                               totale   832.833476
Campania                       femmine  868.322946
                               maschi   738.361796
                               totale   788.892745
Emilia-Romagna                 femmine  392.451681
                               maschi   301.454407
                               totale   341.828237
Friuli-Venezia Giulia          femmine  401.895183
                               maschi   304.199333
                               totale   347.977393
Lazio                          femmine  563.400136
                               maschi   531.508255
                               totale   545.562947
Liguria                        femmine  481.439403
                               maschi   450.097282
                               totale   464.560673
Lombardia                      femmine  359.461552
                               maschi   319.722327
                               totale   337.011343
Marche                         femmine  462.846919
                               maschi   356.422046
                               totale   399.436484
Molise                         femmine  696.154997
                               maschi   598.229409
                               totale   634.277280
Piemonte                       femmine  500.091432
                               maschi   421.814131
                               totale   456.146790
Puglia                         femmine  762.795824
                               maschi   663.060407
                               totale   702.847678
Sardegna                       femmine  747.859626
                               maschi   709.923063
                               totale   725.767942
Sicilia                        femmine  891.240105
                               maschi   767.136200
                               totale   814.476625
Toscana                        femmine  448.976149
                               maschi   357.052491
                               totale   397.502049
Trentino Alto Adige / Südtirol femmine  221.017218
                               maschi   180.071362
                               totale   197.245154
Umbria                         femmine  495.864564
                               maschi   417.764312
                               totale   451.270002
Valle d'Aosta / Vallée d'Aoste femmine  358.377183
                               maschi   346.739865
                               totale   352.299393
Veneto                         femmine  371.132543
                               maschi   252.188610
                               totale   304.724475